test_vm/src/lib.rs

```diff
@@ -284,27 +235,35 @@ where
         self.invocations.take()
     }
 
-    fn actor(&self, address: &Address) -> Option<Actor> {
+    fn actor(&self, address: &Address) -> Option<ActorState> {
         // check for inclusion in cache of changed actors
         if let Some(act) = self.actors_cache.borrow().get(address) {
             return Some(act.clone());
         }
         // go to persisted map
-        let actors =
-            Hamt::<&'bs BS, Actor, BytesKey, Sha256>::load(&self.state_root.borrow(), self.store)
-                .unwrap();
+        let actors = Hamt::<&'bs BS, ActorState, BytesKey, Sha256>::load(
+            &self.state_root.borrow(),
+            self.store,
+        )
+        .unwrap();
         let actor = actors.get(&address.to_bytes()).unwrap().cloned();
         actor.iter().for_each(|a| {
             self.actors_cache.borrow_mut().insert(*address, a.clone());
         });
         actor
     }
 
-    fn actor_manifest(&self) -> BiBTreeMap<Cid, Type> {
-        let actors =
-            Hamt::<&'bs BS, Actor, BytesKey, Sha256>::load(&self.state_root.borrow(), self.store)
-                .unwrap();
-        let mut manifest = BiBTreeMap::new();
+    fn primitives(&self) -> &dyn Primitives {
+        &self.primitives
+    }
+
+    fn actor_manifest(&self) -> BTreeMap<Cid, Type> {
+        let actors = Hamt::<&'bs BS, ActorState, BytesKey, Sha256>::load(
+            &self.state_root.borrow(),
+            self.store,
+        )
+        .unwrap();
+        let mut manifest = BTreeMap::new();
```

Comment:
Can you use the constant ACTOR_TYPES here now? If not, can you leave a comment that this is a slow way of doing this and it should instead be loaded from System actor state like real VM does.